hs/CalendarResponse.hs

```diff
@@ -0,0 +1,285 @@
+        ["ORGANIZER:University of Toronto",
+         "SUMMARY:" ++ code ++ " " ++ sect,
+         "CATEGORIES:EDUCATION", "END:VEVENT"]
+
+-- ** Ordering data
+
+-- | Orders the time fields for each course; which were obtained from the
+-- database
+orderData :: [Time] -> [[[Double]]]
+orderData courseFields = filter (not . null)
+                                (assignDay $ map timeField courseFields)
+
+-- | Organizes the data by the day of the week, assigning a position based on
+-- the day in the time field.
+assignDay :: [[Double]] -> [[[Double]]]
+assignDay lst = [monday, tuesday, wednesday, thursday, friday]
+    where
+        monday = filter (\field -> field !! 0 == 0.0) lst
+        tuesday = filter (\field -> field !! 0 == 1.0) lst
+        wednesday = filter (\field -> field !! 0 == 2.0) lst
+        thursday = filter (\field -> field !! 0 == 3.0) lst
+        friday = filter (\field -> field !! 0 == 4.0) lst
+
+-- ** Start time
+
+-- | Obtains the start times for each course by day.
+startTime :: [[[Double]]] -> [[String]]
+startTime dataInOrder =
+    map (\dataByDay -> map getStrTime
+                           (head (sortList dataByDay) :
+                                   getStartConsecutives (sortList dataByDay)))
+        dataInOrder
+    where
+        sortList dataDay = sort $ map (!! 1) dataDay
+
+-- | Gets the start times that are not the very first start time.
+getStartConsecutives :: [Double] -> [Double]
+getStartConsecutives listTimes =
+    filter (/= 30.0) [if listTimes !! i == listTimes !! (i + 1) - 0.5
+                       then 30.0
+                       else listTimes !! (i + 1)| i <- [0 .. lenForComparison]]
+    where
+        lenForComparison = length listTimes - 2
+
+-- ** End time
+
+-- | Obtains the end times for each course by day.
+endTime :: [[[Double]]] -> [[String]]
+endTime dataInOrder =
+    map (\dataByDay -> map (getStrTime . (+ 0.5))
+                           (getEndConsecutives $ sortList dataByDay ++
+                            [last $ sortList dataByDay]))
+        dataInOrder
+    where
+        sortList dataDay = sort $ map (!! 1) dataDay
+
+-- | Gets the end times that are not the very first end time.
+getEndConsecutives :: [Double] -> [Double]
+getEndConsecutives listTimes =
+    filter (/= 30.0) [if listTimes !! i == listTimes !! (i + 1) - 0.5
+                       then 30.0
+                       else listTimes !! i| i <- [0 .. lenForComparison]]
+    where
+        lenForComparison = length listTimes - 2
+
+-- ** Function that works for both start and end times
+
+-- | Creates a string time in the following way HourMinutesSeccondsZ.
+-- For instance, 133500Z corresponds to 1:35 pm.
+getStrTime :: Double -> String
+getStrTime fullTime =
+                  if maybe True (== 0) minutes
+                  then hour ++ "0000"
+                  else hour ++ maybe "0000" getStrMinutes minutes ++ "00"
+    where
+        hours = splitOn "." (show fullTime)
+        hour = hours !! 0
```

Comment:
I disagree that `head` should be used in general. If the code uses `!!` to access elements of a list at different indices, then using `head` for the first element and `!! #` for the rest of the elements seems inconsistent. HLint typically warns against this, but in their [documentation](http://community.haskell.org/~ndm/darcs/hlint/hlint.htm), they comment on this topic:

> Warning - for example x !! 0 suggests head x as a "warning" severity hint. Typically head is a simpler way of expressing the first element of a list, especially if you are treating the list inductively. However, in the expression f (x !! 4) (x !! 0) (x !! 7), replacing the middle argument with head makes it harder to follow the pattern, and is probably a bad idea. Warning hints are often worthwhile, but should not be applied blindly.

This bit of code does indeed use `!! 1` on the subsequent line.